.github/workflows/go.yml

```diff
@@ -1,22 +1,39 @@
 # This file adheres to the YAML5 style.
 {
   name: "Go",
-  on: ["push", "pull_request"],
+  on: [ "push", "pull_request" ],
   jobs: {
     build: {
       name: "Build",
       "runs-on": "ubuntu-latest",
+      strategy: { matrix: { "go-version": [ "stable", "oldstable"] } },
       steps: [
+        { name: "Check out code into the Go module directory", uses: "actions/checkout@v3" },
         {
-          name: "Set up Go 1.18",
-          uses: "actions/setup-go@v1",
-          "with": {"go-version": 1.18},
+          name: "Set up Go ${{ matrix.go-version }}",
+          uses: "actions/setup-go@v4",
+          with: { "go-version": "${{ matrix.go-version }}" },
           id: "go",
         },
-        {name: "Check out code into the Go module directory", uses: "actions/checkout@v1"},
-        {name: "Linter", run: "make lint"},
-        {name: "Test", run: "make test"},
-        {name: "Test release", run: "make test-release"},
+        { name: "Linter", run: "make lint" },
+        { name: "Test", run: "make test" },
+        { name: "Test release", run: "make test-release" },
+      ],
+    },
+    unstable: {
+      name: "Build unstable",
+      "runs-on": "ubuntu-latest",
+      strategy: { matrix: { "go-version": [ "1.21.0-rc.3"] } },
```

Comment:
```suggestion
      strategy: { matrix: { "go-version": [ "1.21.0-rc.4"] } },
```

Reply:
I already try rc4 on golangci-lint, I'm waiting for feedback https://github.com/quasilyte/go-ruleguard/pull/451#issuecomment-1641846116 before updating.